File: fix-params.js
const fs = require('fs');
const path = require('path');

function getAllFiles(dirPath, arrayOfFiles = []) {
  const files = fs.readdirSync(dirPath);

  files.forEach(file => {
    const filePath = path.join(dirPath, file);
    if (fs.statSync(filePath).isDirectory()) {
      arrayOfFiles = getAllFiles(filePath, arrayOfFiles);
    } else if (file.endsWith('.ts')) {
      arrayOfFiles.push(filePath);
    }
  });

  return arrayOfFiles;
}

function fixParamsInFile(filePath) {
  try {
    const content = fs.readFileSync(filePath, 'utf8');
    const originalContent = content;
    
    // Find destructured params
    const pattern = /const\s+\{\s*([^}]+)\s*\}\s*=\s*await\s+params/g;
    const matches = [...content.matchAll(pattern)];
    
    if (matches.length === 0) return false;
    
    let newContent = content;
    
    matches.forEach(match => {
      const vars = match[1].split(',').map(v => v.trim());
      
      vars.forEach(varStr => {
        let paramName, localName;
        
        if (varStr.includes(':')) {
          const parts = varStr.split(':');
          paramName = parts[0].trim();
          localName = parts[1].trim();
        } else {
          paramName = varStr.trim();
          localName = varStr.trim();
        }
        
        // Replace params.paramName with localName
        const replacePattern = new RegExp(`\\bparams\\.${paramName}\\b`, 'g');
        newContent = newContent.replace(replacePattern, localName);
      });
    });
    
    if (newContent !== originalContent) {
      fs.writeFileSync(filePath, newContent, 'utf8');
      return true;
    }
    
    return false;
  } catch (error) {
    console.error(`Error processing ${filePath}:`, error.message);
    return false;
  }
}

function main() {
  const apiDir = path.join(__dirname, 'src', 'app', 'api');
  const files = getAllFiles(apiDir);
  let fixedCount = 0;
  
  files.forEach(file => {
    if (fixParamsInFile(file)) {
      console.log(`Fixed: ${file}`);
      fixedCount++;
    }
  });
  
  console.log(`\nTotal files fixed: ${fixedCount}`);
}

main();

File: fix-params.py
import os
import re

def fix_params_in_file(filepath):
    """Fix params.id usage in Next.js 15 API routes"""
    try:
        with open(filepath, 'r', encoding='utf-8') as f:
            content = f.read()
        
        original_content = content
        
        # Find all params destructuring patterns
        # Pattern: const { id } = await params or const { slug } = await params etc.
        destructure_pattern = r'const\s+\{\s*([^}]+)\s*\}\s*=\s*await\s+params'
        matches = re.findall(destructure_pattern, content)
        
        if not matches:
            return False
        
        # For each destructured variable, replace params.variable with variable
        for match in matches:
            # Split by comma to handle multiple destructured vars
            vars = [v.strip() for v in match.split(',')]
            
            for var in vars:
                # Handle cases like "id: postId" or just "id"
                if ':' in var:
                    parts = var.split(':')
                    param_name = parts[0].strip()
                    local_name = parts[1].strip()
                else:
                    param_name = var.strip()
                    local_name = var.strip()
                
                # Replace params.paramName with localName
                # Use word boundaries to avoid partial matches
                pattern = r'\bparams\.' + re.escape(param_name) + r'\b'
                content = re.sub(pattern, local_name, content)
        
        # Only write if content changed
        if content != original_content:
            with open(filepath, 'w', encoding='utf-8') as f:
                f.write(content)
            return True
        
        return False
    except Exception as e:
        print(f"Error processing {filepath}: {e}")
        return False

def main():
    """Process all TypeScript files in src/app/api"""
    api_dir = 'src/app/api'
    fixed_count = 0
    
    for root, dirs, files in os.walk(api_dir):
        for file in files:
            if file.endswith('.ts'):
                filepath = os.path.join(root, file)
                if fix_params_in_file(filepath):
                    print(f"Fixed: {filepath}")
                    fixed_count += 1
    
    print(f"\nTotal files fixed: {fixed_count}")

if __name__ == '__main__':
    main()
